netmanager/src/views/components/AirQlouds/AirQloudView.js

```diff
@@ -104,11 +107,14 @@ const AirQloudView = (props) => {
 
     setLoading(true);
 
-    let data = {
-      startDateTime: roundToStartOfDay(new Date(startDate).toISOString()),
-      endDateTime: roundToEndOfDay(new Date(endDate).toISOString()),
-      grid: airqloud?._id
-    };
+    let data = {};
+    if (airqloud) {
+      data = {
+        startDateTime: roundToStartOfDay(new Date(startDate).toISOString()),
+        endDateTime: roundToEndOfDay(new Date(endDate).toISOString()),
+        grid: airqloud._id
```

Comment:
Hi @OchiengPaul442 , thank you so much for this PR.
We might need to have a walkthrough session for this PR since:

1.  I am not seeing how we are now checking for the presence or absence of airqloud._id. Won't this change introduce new runtime errors?
2. Secondly, a recently merged PR for Samantha seems oba to fix things? Most recent deployment has worked well.

Please remember to bring up this PR during any of our syncs or feel free to find any appropriate late evening sync this week where we can discuss the changes here.

Thanks again!

Reply:
Hi @Baalmart 

Thank you for your feedback. Regarding the replacement of **optional chaining operators**, I’ve primarily used **`If statements and ternary operations`** to check for the existence of variables before using them. For instance, before using airquloud._id, I verify its presence; otherwise, I return nothing to prevent errors.
I hope this clarifies my approach.

Additionally, since optional chaining operators are not fully supported in our current Node version 14, this PR replaces them with **`if statements and ternary operations`** for Netmanager. This is done to prevent potential issues in the future.

Reply:
thanks! makes sense!